user/user.go

```diff
@@ -0,0 +1,49 @@
+package user
+
+import (
+	"crypto/sha256"
+	"errors"
+	"regexp"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+var (
+	// EmailValidatorRegEx checks if a string is a valid email address.
+	// See https://emailregex.com/
+	EmailValidatorRegEx = regexp.MustCompile(`^[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,64}$`)
+
+	// ErrInvalidEmail is returned when we encounter an invalid email value.
+	ErrInvalidEmail = errors.New("invalid email")
+)
+
+type (
+	// Email is an email.
+	Email string
+
+	// Hash represents a 256bit hash value.
+	Hash [32]byte
+
+	// User represents a Skynet user.
+	User struct {
+		// ID is a hexadecimal string representation of the MongoDB id assigned
+		// to this user object. It is auto-generated by Mongo on insert.
+		ID        primitive.ObjectID `bson:"_id"`
+		FirstName string             `bson:"firstName" json:"firstName"`
+		LastName  string             `bson:"lastName" json:"lastName"`
+		Email     Email              `bson:"email" json:"email"`
```

Comment:
I think to keep it simple we might only need a first and last name for the MVP

Reply:
Having the email here is critical, IMO, because it is going to be our unique registration identifier, i.e. your username. Otherwise we can give people their internal IDs but those are hardly user-friendly and also I'd rather not expose those at all as they are implementation-depenedent.

Additionally, having emails here and requiring account verification via email will prevent a lot of potential abuse via automatic creation of bogus accounts in order to go around the speed limits.

Reply:
Right I actually meant "not" instead of "only" ha. So only email and password should be fine in the beginning I think.

Reply:
Oh, right, yes. Those are optional anyway - the email is the only required field.